pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -437,11 +445,38 @@ func TestPlugin(t *testing.T) {
 			pod:    podWithClaimName,
 			claims: []*resourcev1alpha2.ResourceClaim{inUseClaim},
 		},
+		"disable": {
+			pod:    podWithClaimName,
+			claims: []*resourcev1alpha2.ResourceClaim{inUseClaim},
+			want: want{
+				prefilter: result{
+					status: framework.NewStatus(framework.Skip),
+				},
+			},
+		},
 	}
 
 	for name, tc := range testcases {
 		// We can run in parallel because logging is per-test.
 		tc := tc
+		if name == "disable" {
+			t.Run(name, func(t *testing.T) {
+				t.Parallel()
+				nodes := tc.nodes
+				if nodes == nil {
+					nodes = []*v1.Node{workerNode}
+				}
+				testCtx := setup(t, nodes, tc.claims, tc.classes, tc.schedulings)
+				testCtx.p.enabled = false
+				initialObjects := testCtx.listAll(t)
+				result, status := testCtx.p.PreFilter(testCtx.ctx, testCtx.state, tc.pod)
+				t.Run("prefilter", func(t *testing.T) {
+					assert.Equal(t, tc.want.preFilterResult, result)
+					testCtx.verify(t, tc.want.prefilter, initialObjects, result, status)
+				})
+			})
+			continue
+		}
```

Comment:
This complicates the test code.
Can't we somehow run "disable" testcase with the common `t.Run` below

Reply:
Optimized the code, PTAL, thanks.